src/application.vala

```diff
@@ -0,0 +1,109 @@
+/* application.vala
+ *
+ * Copyright 2022 Allie Law
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+namespace Mixer {
+    public class Application : Adw.Application {
+
+        private static bool print_version = false;
+        private static string mockup = "";
+        private Adw.ApplicationWindow win;
+
+        private const GLib.OptionEntry[] OPTIONS = {
+
+            { "version", '\0', OptionFlags.NONE, OptionArg.NONE, ref print_version,
+            "Display version number", null },
+            { "mockup", 'm', OptionFlags.NONE, OptionArg.STRING, ref mockup},
+            // list terminator
+            { null }
+        };
+
+        private Settings settings = new Settings (Constants.APP_ID);
+
+
+        public Application () {
+            Object (
+                application_id: Constants.APP_ID,
+                flags: ApplicationFlags.FLAGS_NONE
+            );
+
+            this.add_main_option_entries (OPTIONS);
+        }
+
+
+        construct {
```

Comment:
You need the following code somewhere in the constructor of this class to show your app in the user language:

```vala
            Intl.setlocale (LocaleCategory.ALL, "");
            Intl.bindtextdomain (Constants.GETTEXT_PACKAGE, Constants.LOCALE_DIR);
            Intl.bind_textdomain_codeset (Constants.GETTEXT_PACKAGE, "UTF-8");
            Intl.textdomain (Constants.GETTEXT_PACKAGE);
```